xarray/core/datatree_render.py

```diff
@@ -0,0 +1,252 @@
+"""
+String Tree Rendering. Copied from anytree.
+
+Minor changes to `RenderDataTree` include accessing `children.values()`, and
+type hints.
+
+"""
+
+from __future__ import annotations
+
+from collections import namedtuple
+from collections.abc import Iterable, Iterator
+from typing import TYPE_CHECKING
+
+if TYPE_CHECKING:
+    from xarray.core.datatree import DataTree
+
+Row = namedtuple("Row", ("pre", "fill", "node"))
+
+
+class AbstractStyle:
+    def __init__(self, vertical: str, cont: str, end: str):
+        """
+        Tree Render Style.
+        Args:
+            vertical: Sign for vertical line.
+            cont: Chars for a continued branch.
+            end: Chars for the last branch.
+        """
+        super().__init__()
+        self.vertical = vertical
+        self.cont = cont
+        self.end = end
+        assert (
+            len(cont) == len(vertical) == len(end)
+        ), f"'{vertical}', '{cont}' and '{end}' need to have equal length"
+
+    @property
+    def empty(self) -> str:
+        """Empty string as placeholder."""
+        return " " * len(self.end)
+
+    def __repr__(self) -> str:
+        return f"{self.__class__.__name__}()"
+
+
+class ContStyle(AbstractStyle):
+    def __init__(self):
+        """
+        Continued style, without gaps.
+
+        >>> from xarray.core.datatree import DataTree
+        >>> from xarray.core.render import RenderDataTree
+        >>> root = DataTree(name="root")
+        >>> s0 = DataTree(name="sub0", parent=root)
+        >>> s0b = DataTree(name="sub0B", parent=s0)
+        >>> s0a = DataTree(name="sub0A", parent=s0)
+        >>> s1 = DataTree(name="sub1", parent=root)
+        >>> print(RenderDataTree(root))
+
+        DataTree('/root', parent=None)
+        ├── DataTree('sub0')
+        │   ├── DataTree('sub0B')
+        │   └── DataTree('sub0A')
+        └── DataTree('sub1')
+        """
+        super().__init__("\u2502   ", "\u251c\u2500\u2500 ", "\u2514\u2500\u2500 ")
+
+
+class RenderDataTree:
+    def __init__(
+        self,
+        node: DataTree,
+        style=ContStyle(),
+        childiter: type = list,
+        maxlevel: int | None = None,
+    ):
+        """
+        Render tree starting at `node`.
+        Keyword Args:
+            style (AbstractStyle): Render Style.
+            childiter: Child iterator. Note, due to the use of node.children.values(),
+                Iterables that change the order of children  cannot be used
+                (e.g., `reversed`).
+            maxlevel: Limit rendering to this depth.
+        :any:`RenderDataTree` is an iterator, returning a tuple with 3 items:
+        `pre`
+            tree prefix.
+        `fill`
+            filling for multiline entries.
+        `node`
+            :any:`NodeMixin` object.
+        It is up to the user to assemble these parts to a whole.
+        >>> from xarray import Dataset
```

Comment:
The examples in this documentation string are a bit shorter than the originals from `anytree`. That's because using `node.children.values()` gets a `ValuesView` which isn't compatible with iterables like `reversed` that alter the order of the items in the iterable.